src/App.tsx

```diff
@@ -1,14 +1,66 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from './app/hooks';
+import { actions } from './features/todos';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const dispatch = useDispatch();
+  const { todos } = useAppSelector(state => state.todos);
+  const [showLoader, setShowLoader] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { query } = useAppSelector(state => state.filter);
+
+  const filter = useAppSelector(state => state.filter);
+
+  useEffect(() => {
+    setShowLoader(true);
+    getTodos()
+      .then(response => {
+        dispatch(actions.setTodos(response));
+      })
+      .finally(() => setShowLoader(false));
+  }, [dispatch]);
+
+  useMemo(() => {
+    if (todos) {
+      setVisibleTodos(todos);
+    }
+  }, [todos]);
```

Comment:
we don't use useMemo like that, you can replace it with useEffect

but we can simplify it and remove the visibleTodos state at all
just create the var
`const visibleTodos = getVisibleTodos(...)`

https://react.dev/reference/react/useMemo#usage

Reply:
fixed)